athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -251,8 +253,14 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
             LOGGER.info("MetadataHandler=GcsMetadataHandler|Method=doGetSplits|Message=StorageSplit JSON\n{}",
                     storageSplitJson);
             Split.Builder splitBuilder = Split.newBuilder(spillLocation, makeEncryptionKey())
-                    .add(CLASSIFICATION_GLUE_TABLE_PARAM, table.getParameters().get(CLASSIFICATION_GLUE_TABLE_PARAM))
+                    .add(FILE_FORMAT, table.getParameters().get(CLASSIFICATION_GLUE_TABLE_PARAM))
                     .add(STORAGE_SPLIT_JSON, storageSplitJson);
+
+            // set partition column name and value in split
+            for (FieldVector fieldVector : partitions.getFieldVectors()) {
+                fieldVector.getReader().setPosition(curPartition);
+                splitBuilder.add(fieldVector.getName(), fieldVector.getReader().readObject().toString());
```

Comment:
I would also add a check here to make sure that none of the `fieldVector.getName()`s are the same as `FILE_FORMAT`, since they are both just top level keys.

If they are the same, just throw a RuntimeException to stop us from continuing.

Reply:
updated the code.